pkg/controllers/disruption/helpers.go

```diff
@@ -219,7 +219,12 @@ func BuildDisruptionBudgets(ctx context.Context, cluster *state.Cluster, clk clo
 			continue
 		}
 		nodePool := node.Labels()[v1beta1.NodePoolLabelKey]
-		if node.MarkedForDeletion() {
+		// If the node satisfies one of the following, we subtract it from the allowed disruptions.
+		// 1. Has a NotReady conditiion
+		// 2. Is marked as deleting
+		if _, found := lo.Find(node.Node.Status.Conditions, func(nc v1.NodeCondition) bool {
```

Comment:
I'm trying to remember, but I thought that there was a helper function already for this check